cmake/LLVMCheck.cmake

```diff
@@ -1,7 +1,8 @@
 if((CMAKE_CXX_COMPILER_ID MATCHES "[Cc]lang") OR
    (CMAKE_CXX_COMPILER_ID MATCHES "IntelLLVM"))
-  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.0.0)
-    message(FATAL_ERROR "this project requires clang >= 8.0")
+
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 17.0.0)
+    set(CLANG_VERSION_LESS_17 ON)
```

Comment:
Why have different macros for "LESS"? Why not just have CLANG_MAJOR_VERSION and check against it?